airflow/docker_image.go

```diff
@@ -353,27 +382,40 @@ func (d *DockerImage) Push(registry, username, token, remoteImage string) error
 	if err != nil {
 		log.Debugf("Error pushing image to docker: %v", err)
 		// if NewClientWithOpt does not work use bash to run docker commands
-		return useBash(&authConfig, remoteImage)
+		return err
 	}
 	defer responseBody.Close()
-	err = displayJSONMessagesToStream(responseBody, nil)
+	return displayJSONMessagesToStream(responseBody, nil)
+}
+
+// Get the registry name to authenticate against
+func (d *DockerImage) getRegistryToAuth(imageName string) (string, error) {
+	domain, err := config.GetCurrentDomain()
 	if err != nil {
-		return useBash(&authConfig, remoteImage)
+		return "", err
 	}
-	// Delete the image tags we just generated
-	err = cmdExec(dockerCommand, nil, nil, "rmi", remoteImage)
-	if err != nil {
-		return fmt.Errorf("command '%s rmi %s' failed: %w", dockerCommand, remoteImage, err)
+
+	if domain == "localhost" {
+		return config.CFG.LocalRegistry.GetString(), nil
 	}
-	return nil
+	parts := strings.SplitN(imageName, "/", 2)
```

Comment:
could we try this change out with a custom registry on Software, not sure if it will break or not @rujhan-arora-astronomer @rishkarajgi 
with a custom registry, our image name would be of the form `fmt.Sprintf("%s:%s", registry, fmt.Sprintf("%s-%s", name, tag))`

Reply:
Oh yeah, there's a good chance that this won't work in that case.

Reply:
We already have the domain here, so we could look if we are in custom registry mode and return.... `registry`? in that case?

Reply:
If I recollect correctly, the reason we had such a structure was that some of the custom registry endpoints were of the form: `custom-registry.com/dev-airflow`, and it won't allow us to append `/<image_name>` after that.

The only part I am uncertain about is whether trying to log in to just `custom-registry.com` would work or not.

Reply:
Wait a sec, for Software we don't pass username, so we won't try to login during image push, so I think we are good 🙂.
For software either we log in to the registry during `astro login` or we expect the customer to log in manually for the custom registry.

Reply:
Oh yeah!

The login has to be to `custom-registry.com` (i.e. without the repo, but it's not _always_ a bare hostname, it could be `internal.company.com/registry` for instance)


So we're good to merge this?

Reply:
yup